src/py/aspen/app/views/tests/test_phylo_tree_rename.py

```diff
@@ -0,0 +1,94 @@
+import json
+
+from aspen.app.views.phylo_trees import _process_phylo_tree
+from aspen.database.models import CanSee, DataType
+from aspen.test_infra.models.phylo_tree import phylotree_factory
+from aspen.test_infra.models.sample import sample_factory
+from aspen.test_infra.models.usergroup import group_factory, user_factory
+
+
+def test_phylo_tree_rename(session, mock_s3_resource, test_data_dir):
+    """Create a set of samples belonging to different groups with different levels of
+    can-see relationships.  Rename the nodes according to the can-see rules, and verify
+    that the nodes are renamed correctly."""
+    viewer_group = group_factory()
+    can_see_group = group_factory("can_see", email="can_see@yahoo.com")
+    wrong_can_see_group = group_factory(
+        "wrong_can_see", email="wrong_can_see@hotmail.com"
+    )
+    no_can_see_group = group_factory("no_can_see", email="no_can_see@aol.com")
```

Comment:
these are some retro test email accounts lol

Reply:
I was trying to think what else I could think of.  juno.com?  I don't know what else is super old.